apps/playground-sveltekit/src/lib/next-auth.ts

```diff
@@ -0,0 +1,105 @@
+import type { RequestEvent } from '@sveltejs/kit';
+import type { IncomingRequest, NextAuthOptions, Session } from 'next-auth';
+import type { NextAuthAction } from 'next-auth/lib/types';
+import type { OutgoingResponse } from 'next-auth/core';
+import { NextAuthHandler } from 'next-auth/core';
+import cookie from 'cookie';
+import getFormBody from './utils/get-form-body';
+
+async function toSvelteKitResponse(request: Request, nextAuthResponse: OutgoingResponse<unknown>) {
+  const { headers, cookies, body, redirect, status = 200 } = nextAuthResponse;
+
+  const response = {
+    status,
+    headers: {}
+  };
+
+  headers?.forEach((header) => {
+    response.headers[header.key] = header.value;
+  });
+
+  response.headers['set-cookie'] = cookies?.map((item) => {
+    return cookie.serialize(item.name, item.value, item.options);
+  });
+
+  if (redirect) {
+    let formData = null;
+    try {
+      formData = await request.formData();
+      formData = getFormBody(formData);
+    } catch {
+      // no formData passed
+    }
+    if (formData?.json !== 'true') {
+      response.status = 302;
+      response.headers['Location'] = redirect;
+    } else {
+      response['body'] = { url: redirect };
+    }
+  } else {
+    response['body'] = body;
+  }
+
+  return response;
+}
+
+async function SKNextAuthHandler({ request, url, params }: RequestEvent, options: NextAuthOptions) {
+  const nextauth = params.nextauth.split('/');
+  let body = null;
+  try {
+    body = await request.formData();
+    body = getFormBody(body);
+  } catch {
+    // no formData passed
+  }
+
```

Comment:
```suggestion
  options.secret = import.meta.env.VITE_NEXTAUTH_SECRET
```

Do you know if third-party libraries can read environment variables like this?

This would let the user configure fewer things. We would magically inject this value if they named the env var as we expected here.

Reply:
I think only vite reads this